tests/plugins/test_import_image.py

```diff
@@ -188,6 +189,35 @@ def test_import_image_retry(monkeypatch):
     runner.run()
 
 
+def test_import_image_retry_and_fail(monkeypatch):
+    """
+    Test retrying importing tags for an existing ImageStream
+    """
+
+    runner = prepare(retry_delay=1, retry_attempts=3)
+
+    build_json = {"metadata": {}}
+    monkeypatch.setenv("BUILD", json.dumps(build_json))
+
+    (flexmock(OSBS)
+     .should_receive('get_image_stream')
+     .once()
+     .with_args(TEST_IMAGESTREAM))
+    (flexmock(OSBS)
+     .should_receive('create_image_stream')
+     .never())
+    (flexmock(OSBS)
+     .should_receive('import_image')
+     .times(3)
+     .with_args(TEST_IMAGESTREAM)
+     .and_return(False)
```

Comment:
Maybe a test with:
```python
.and_return(False)
.and_return(False)
.and_return(True)
```
to ensure retry is attempted and plugin works as expected?